### PR TITLE
Resubmit #29747. "Add RpcAgentTestFixture to extract duplicate code"

### PR DESCRIPTION
**Pull request by xush6528**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30093 Resubmit "Add `RpcAgentOptions` struct type, which bundles different required arguments for different `RpcAgent`s"
* **#30092 Resubmit "Add RpcAgentTestFixture to extract duplicate code"**

There are duplicate code for component that rely on RpcAgent. Extract them into a re-usable test fixture class.

Differential Revision: [D18595408](https://our.internmc.facebook.com/intern/diff/D18595408/)

### CODE REVIEW

File: test/dist_optimizer_test.py [hunk 1]
```diff
@@ -2,13 +2,14 @@
...
 
-from dist_utils import INIT_METHOD_TEMPLATE, dist_init
+from dist_utils import dist_init
 from torch import optim
 from torch.distributed.optim import DistributedOptimizer
 import torch
 import torch.distributed.autograd as dist_autograd
 import torch.distributed.rpc as rpc
 import threading
+from rpc_agent_test_fixture import RpcAgentTestFixture
```
Comment (mrshenli) [inline test/dist_optimizer_test.py]: move from up to the front of import?

File: test/rpc_test.py [hunk 2]
```diff
@@ -12,14 +12,16 @@
 import torch.distributed.rpc as rpc
 from torch.distributed.rpc import RRef
 from common_utils import load_tests
-from dist_utils import INIT_METHOD_TEMPLATE, TEST_CONFIG, dist_init
+import dist_utils
+from dist_utils import dist_init
 from torch.distributed.rpc.internal import PythonUDF, _internal_rpc_pickler
+from rpc_agent_test_fixture import RpcAgentTestFixture
```
Comment (mrshenli) [inline test/rpc_test.py]: let's sort the import section

**Review by mrshenli — APPROVED** [review]
Approving as already reviewed the previous PR.

@xush6528 would I be correct if I assume the only difference here is getting rid of changes in `run_test.py`?

File: test/dist_optimizer_test.py (hunk 1 above)
Comment (xush6528) [reply test/dist_optimizer_test.py]: Let me do a whole formatting in a separate PR.

**Comment by xush6528** [comment]
@mrshenli Yes. Removed the changes to `run_test.py`, which is not necessary to the purpose of this PR, and is of high risk.

File: test/dist_optimizer_test.py (hunk 1 above)
Comment (mrshenli) [reply test/dist_optimizer_test.py]: sure

**Comment by xush6528** [comment]
Redoing https://github.com/pytorch/pytorch/pull/29747.

**Pull request closed without merging**